courses/utils_test.py

```diff
@@ -29,6 +33,94 @@ def program():
     return ProgramFactory.create()
 
 
+@pytest.fixture()
+def course():
+    """Course object fixture"""
+    return CourseFactory.create()
+
+
+def test_course_run_certificate(user, course):
```

Comment:
Please combine some of these test cases with parameters, and get rid of some of the repeated testing logic. Some of this logic may be a little off, so please adjust as necessary

```python
@pytest.mark.parametrize("grade,passed,exp_created", [[0.25, True, True], [0.0, True, False], [1.0, False, False]])
def test_process_course_run_certificate(user, course, grade,passed, exp_created):
    grade = CourseRunGradeFactory.create(
        course_run__course=course, user=user, grade=grade, passed=passed
    )
    certificate, created, deleted = process_course_run_grade_certificate(grade)
    assert created is exp_created
    assert bool(certificate) is exp_created
    assert not deleted

def test_course_run_certificate_idempotent(user, course):
    grade = CourseRunGradeFactory.create(
        course_run__course=course, user=user, grade=0.25, passed=True
    )
    process_course_run_grade_certificate(grade)
    certificate, created, deleted = process_course_run_grade_certificate(grade)
    assert certificate is not None
    assert created is False
    assert deleted is False

def test_course_run_certificate_blacklist(user, course):
    grade = CourseRunGradeFactory.create(
        course_run__course=course, user=user, grade=0.25, passed=True
    )
    process_course_run_grade_certificate(grade)
    blacklist_grade = CourseRunGradeFactory.create(
        course_run__course=course, user=user, grade=0.0, passed=False
    )
    certificate, created, deleted = process_course_run_grade_certificate(blacklist_grade)
    assert certificate is None
    assert not created
    assert not deleted
```

Reply:
Updated